docs: Update README, env file with Boot tests and static user (HMS-9200) (#3688)
RedHatInsights/image-builder-frontend: README.md

5. Now you have two options of how to run the tests:
   * (Preferred) Use VS Code and the [Playwright Test module for VSCode](https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright). But other editors do have similar plugins for ease of use, if so desired
   * Using terminal - `npx playwright test` will run the playwright test suite. `npx playwright test --headed` will run the suite in a vnc-like browser so you can watch it's interactions.

### Dynamic vs static users

> This applies on **stage** environment only. By default we generate new user for each CI run for the hosted tests, however there is an option to use static user as well. You can set credentials for said static user like this:

Reviewer: ```suggestion
This applies to the **stage** environment only. By default we generate a new user for each CI run for the hosted tests, however there is an option to use static user as well. You can set credentials for said static user like this:
```

```.env
PLAYWRIGHT_STATIC_USER="<your_static_user_username>"
PLAYWRIGHT_STATIC_PASSWORD="<your_static_user_password>"
```
For local development purposes, you can use the same credentials as for `PLAYWRIGHT_USER` and `PLAYWRIGHT_PASSWORD` if you are using your stage account for those.
## Playwright Boot tests

> This section describes what are Playwright Boot tests, how they work and how to run them locally.

Reviewer: ```suggestion
This section describes what Playwright Boot tests are, how they work and how to run them locally.
```

Boot tests provide end to end coverage for Image Builder and they are used to test mainly integrations with other services. Their main advantage is that they build a image, upload it and **launch it on RHOSP** (RedHat OpenStack Platform). This way we can test images and their customizations through remotely executed commands on an actual running VM booted from the image.
Boot tests are located in the [playwright/BootTests](playwright/BootTests) directory.

### Local development setup
In order to run the Boot tests locally, we need to set up few things first on top of what we done in [Running hosted service Playwright tests](#running-hosted-service-playwright-tests) section.

We need additional fields in the .env file, some of them are already set and don't need to be changed in the [example env file](playwright_example.env)), but some of them have to be set manually, specifically following:

```.env
OS_APPLICATION_CREDENTIAL_ID=<your_id>
OS_APPLICATION_CREDENTIAL_SECRET=<your_secret>
OS_SSH_KEY_NAME="<name_of_your_ssh_key_entry_in_openstack>"
```

> In order to be able to acces RHOSP within the Boot tests, we need to generate credentials and create an entry with our public ssh key on the platform. Log into [RHOSP dashboard](https://api.rhos-01.prod.psi.rdu2.redhat.com/) using **Keystone credentials** and navigate to Identity -> Applicaiton Credentials. There you can create the credentials (you will get values for `OS_APPLICATION_CREDENTIAL_ID` and `OS_APPLICATION_CREDENTIAL_SECRET`). In order to create an entry for your SSH key, navigate to Project -> Compute -> Keys and add your public key there.

Reviewer: ```suggestion
In order to be able to access RHOSP within the Boot tests, we need to generate credentials and create an entry with our public ssh key on the platform. Log into [RHOSP dashboard](https://api.rhos-01.prod.psi.rdu2.redhat.com/) using **Keystone credentials** and navigate to Identity -> Application Credentials. There you can create the credentials (you will get values for `OS_APPLICATION_CREDENTIAL_ID` and `OS_APPLICATION_CREDENTIAL_SECRET`). In order to create an entry for your SSH key, navigate to Project -> Compute -> Keys and add your public key there.
```

> By filling out these variables you should be able to run the Boot test locally successfuly.

Reviewer: ```suggestion
By filling out these variables you should be able to run the Boot test locally successfully.
```

### CI setup
Boot tests run as a scheduled nightly Github Action with custom runner in AWS Codebuild, but there is an option to run them manually on a PR as well.
#### Manual run
In order to run the Boot tests on a PR, you can open the `Boot tests` workflow in Github Actions and type in the number of PR into a `Pull Request number to run tests against` field. The action will then pull the code from a PR of that number and execute as usual.

#### Where can I see results?

> You can find artifacts and link to a Currents report directly in the workflow run detail, but Currents will also link the report to back to the PR when the workflow finishes as a *check*.

Author: Whoa, that was lot of typos, thanks for correcting this ! :tada:
